strapi/strapi
# upload-aws-s3-path-prefix

### What does it do?

It allows to add a `pathPrefix` to the `upload-aws-s3` provider. And suggests the move of all other options to be passed via the s3 property.  If there is no s3 property given, the provider still passes all the given properties to the `AWS.S3` constructor.

lodash depdendency removed, cause i could not find a use of it.

### Why is it needed?

Currently its not possible (at least not in a way that prevent conflicts) to share a bucket over multiple projects for example strapi installations. As far as i can tell the S3 client (nor the spec) support some prefixing out of the box, so if it is wished, it would be needed to be done in the project itself, or in a third party s3 provider (which i would like to prevent, cause the rest is the same).

### How to test it?

Set the property `pathPrefix` to '' or undefined or null => nothing should change
Set the property `pathPrefix` to for example 'foo', every file should be uploaded and deleted from a path starting with foo/path/to/file.
